### ISSUE TITLE
Crash when mouse tracking is enabled

### ISSUE THREAD

**Issue by cortesi** [urwid/urwid#104]
Hi,

I'm working on enabling mouse scrolling in mitmproxy's console app. After enabling mouse input, a crash is triggered:

```
Traceback (most recent call last):
  File "/Users/cortesi/mitmproxy/mitmproxy/libmproxy/console/__init__.py", line 611, in run
    self.ui.run_wrapper(self.loop)
  File "/Library/Python/2.7/site-packages/urwid/display_common.py", line 757, in run_wrapper
    return fn()
  File "/Users/cortesi/mitmproxy/mitmproxy/libmproxy/console/__init__.py", line 825, in loop
    keys = self.ui.get_input()
  File "/Library/Python/2.7/site-packages/urwid/raw_display.py", line 324, in get_input
    keys, raw = self.parse_input(None, None, self.get_available_raw_input())
  File "/Library/Python/2.7/site-packages/urwid/raw_display.py", line 465, in parse_input
    codes, wait_for_more)
  File "/Library/Python/2.7/site-packages/urwid/escape.py", line 383, in process_keyqueue
    if run[0] == "esc" or run[0].find("meta ") >= 0:
AttributeError: 'tuple' object has no attribute 'find'
```

You can reproduce this as follows: 
- Install the "mouse" branch of mitmproxy
- Start mitmproxy
- Press "Escape", and then click anywhere on the screen

This issue does not occur when the set_mouse_tracking call is removed.

I've tested this with iTerm on OSX Yosemite.

Cheers,

Aldo

**Comment by wardi** [comment]
this might just be one of urwid's sharp edges. You can't pass all get_input return values to things only expecting keyboard input.

Can you link to the source that you're testing against? Or better yet provide a simple example that reproduces the problem?

**Comment by cortesi** [comment]
Hi there. I see this similar crashes with scripts in the urwid examples directory, e.g. pop_up.py. Here, I checked out and installed the current urwid master, ran pop_up.py, then alternated pressing escape and clicking the mouse:


```
Traceback (most recent call last):
  File "./pop_up.py", line 40, in <module>
    loop.run()
  File "/Library/Python/2.7/site-packages/urwid-1.3.1.dev0-py2.7-macosx-10.10-intel.egg/urwid/main_loop.py", line 278, in run
    self._run()
  File "/Library/Python/2.7/site-packages/urwid-1.3.1.dev0-py2.7-macosx-10.10-intel.egg/urwid/main_loop.py", line 379, in _run
    raise e
AttributeError: 'tuple' object has no attribute 'find'

```

**Comment by ulidtko** [comment]
Referencing #188 as it's an obvious instance of this bug, too.

**Comment by and3rson** [comment]
Fixed in #304.
